pkg/ingester/ingester.go

```diff
@@ -321,32 +348,37 @@ func (i *Ingester) append(ctx context.Context, sample *model.Sample) error {
 	i.stopLock.RLock()
 	defer i.stopLock.RUnlock()
 	if i.stopped {
-		return fmt.Errorf("ingester stopping")
+		return created, fmt.Errorf("ingester stopping")
 	}
 
 	i.userStatesMtx.RLock()
 	defer i.userStatesMtx.RUnlock()
-	state, fp, series, err := i.userStates.getOrCreateSeries(ctx, sample.Metric)
+	created, state, fp, series, err := i.userStates.getOrCreateSeries(ctx, sample.Metric)
 	if err != nil {
-		return err
+		return created, err
 	}
 	defer func() {
 		state.fpLocker.Unlock(fp)
 	}()
 
+	if created {
+		userID, _ := user.ExtractOrgID(ctx)
+		i.memSeriesTotal.WithLabelValues(userID).Inc()
```

Comment:
Probably can't include userid in the label - too high cardinality.

Also, if we're just using the `created` return value to increment a metric, perhaps put the metric in the user_state.go and avoid the plumbing?

Reply:
I've thought about it and we can. We're just adding one series per user per ingester. And even a small user will send us 100K metrics, hence the increase in cardinality is insanely low.

Reply:
We are also using `created` to log to the tracing span.

Reply:
> We are also using created to log to the tracing span.

This can be done by passing ctx down to the function, and is probably the preferred way.  Too many return values from a function is a bit of a code smell.

Reply:
The problem is that the `append` is in the critical path and we don't want to add spans here. So I am logging the total number of series created then. For that I'd need the total series created in a single append call which requires the plumbing.

Though I'm curious about the benefit of having `series_created` in the span.

Reply:
Okay let me give this some more though; keeping this code tidy is important.